myriad-scheduler/src/main/java/com/ebay/myriad/scheduler/event/handlers/ResourceOffersEventHandler.java

```diff
@@ -105,11 +106,20 @@ public void onEvent(ResourceOffersEvent event, long sequence,
         for (NodeTask nodeTask : nodeTasks) {
           nodeTask.setSlaveAttributes(offer.getAttributesList());
         }
+        // keep this in case SchedulerState gets out of sync. This should not happen with 
+        // synchronizing addNodes method in SchedulerState
+        // but to keep it safe
+        final Set<Protos.TaskID> blacklistedTasks = Sets.newHashSet();
```

Comment:
It removes from taskIds as well: public synchronized void removeTask(Protos.TaskID taskId)
The only reason it may be OK to remove in place is that we return a copy of pending tasks